src/com/company/Main.java

```diff
@@ -1,46 +1,82 @@
-            String filePathAndName = filePathScanner.nextLine();
-            System.out.println("you entered path: " + filePathAndName + "\n");
-            fileToRead = new File(filePathAndName);
+    private static String getFirstParam (String messageUserCanSeeInConsole, Scanner scanner) {
+        System.out.println(messageUserCanSeeInConsole);
+        return scanner.nextLine();
+    }
+    private static String getDirectoryToSave(String messageUserCanSeeInConsole, Scanner scanner){
+        System.out.println(messageUserCanSeeInConsole);
+        return scanner.nextLine();
+    }
+    private static String getFileNameToSave(String messageUserCanSeeInConsole, Scanner scanner){
+        System.out.println(messageUserCanSeeInConsole);
+        return new StringBuilder().append(scanner.nextLine()).append(".txt").toString();
+    }
+
+    private static void validateFiles(String fileNameToRead, String directoryNameToSaveSortedData, String fileNameToSaveSortedData){
+        try {
+            fileToRead = new File(fileNameToRead);
+            fileToSave = new File(directoryNameToSaveSortedData, fileNameToSaveSortedData);
 
-            System.out.println("Enter file name for saving sorted numbers ");
-            Scanner fileSaveScanner = new Scanner(System.in);
-            String fileSaveSorted = fileSaveScanner.nextLine();
-            fileToSave = new File(fileSaveSorted + ".txt");
+            boolean result = fileToSave.createNewFile();
+            if(result){
+                System.out.println("File to save sorted numbers is created: "+fileToSave.getCanonicalPath());
+            }
+            else {
+                System.out.println("File already exist at location: "+fileToSave.getCanonicalPath());
+            }
 
+            if (!fileToRead.exists()) {
+                System.out.println("\'" + fileNameToRead + "\' file is not found. Check path and file name");
+                System.exit(0);
+            }
+            if (!fileToRead.canRead()) {
+                System.out.println("\'" + fileNameToRead + "\' file impossible to read file. Make another readable file with *.txt extension");
+                System.exit(0);
+            }
+            if (!fileToSave.exists()) {
+                System.out.println("\'" + fileNameToSaveSortedData + "\' file for saving sorted numbers is not created. File \"sortedNumbers\" will be created.");
+                fileToSave = new File("./results/sortedNumbers.txt ");
+            }
+            if(!fileToSave.canWrite()){
+                System.out.println("\'" + fileNameToSaveSortedData + "\' file is invalid for writing");
+            }
         }
         catch (Exception e){
-            System.out.println(e.getMessage());
-            e.printStackTrace();
-        }
-
-        if( FileUtils.isFileExists()){
-            FileUtils.getUnsortedNumbers(fileToRead);
+            try(Writer w = new FileWriter("file.log", true)) {
```

Comment:
Remove this. Save stack to file in another commit

Reply:
removed